packages/api/src/account/service.ts

```diff
@@ -125,8 +126,50 @@ const leaveArchive = async ({
     return deleteResult.rows[0];
   });
 
+const getAccountArchive = async (
+  archiveId: string,
+  email: string
+): Promise<GetAccountArchiveResult | undefined> => {
+  const accountArchiveResult = await db
+    .sql<GetAccountArchiveResult>("account.queries.get_account_archive", {
+      archiveId,
+      email,
+    })
+    .catch((err) => {
+      logger.error(err);
+      throw new createError.InternalServerError(
+        "Failed to retrieve account archive"
+      );
+    });
+  return accountArchiveResult.rows[0];
+};
+
+const getCurrentAccountArchive = async (
+  email: string
+): Promise<GetCurrentAccountArchiveResult> => {
+  const currentAccountArchiveResult = await db
+    .sql<GetCurrentAccountArchiveResult>(
+      "account.queries.get_current_account_archive",
+      { email }
+    )
+    .catch((err) => {
+      logger.error(err);
+      throw new createError.InternalServerError(
+        "Failed to retrieve current account archive"
+      );
+    });
+  if (!currentAccountArchiveResult.rows[0]) {
+    throw new createError.InternalServerError(
+      "Current account archive not found"
+    );
+  }
+  return currentAccountArchiveResult.rows[0];
```

Comment:
The query here could be returning multiple rows, and all of them matter to us, since we want to check whether the folder is shared with _any_ archive the caller is a member of.